File: libs/power-log-parser/src/lib/state/combined-state.ts
import { ParserState, StateType, INodeParser } from './parser-state';
import { StateFacade } from './state-facade';

export class CombinedState {
	GSState!: ParserState;
	PTLState!: ParserState;
	StateFacade!: StateFacade;

	constructor(createNodeParser: (stateFacade: StateFacade, stateType: StateType) => INodeParser) {
		this.StateFacade = new StateFacade(this);
		const gsNodeParser = createNodeParser(this.StateFacade, StateType.GameState);
		const ptlNodeParser = createNodeParser(this.StateFacade, StateType.PowerTaskList);
		this.GSState = new ParserState(StateType.GameState, gsNodeParser, this.StateFacade);
		this.PTLState = new ParserState(StateType.PowerTaskList, ptlNodeParser, this.StateFacade);
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Parser/CombinedState.cs
using HearthstoneReplays.Events;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HearthstoneReplays.Parser
{
    public class CombinedState
    {
        // The "future" state, before the animations happen
        public ParserState GSState;
        // The "real-time" state
        public ParserState PTLState;
        public StateFacade StateFacade;

        public CombinedState()
        {
            StateFacade = new StateFacade(this);
            EventQueueHandler handler = new EventQueueHandler(StateFacade);
            GSState = new ParserState(StateType.GameState, handler, StateFacade);
            PTLState = new ParserState(StateType.PowerTaskList, handler, StateFacade);
        }

        //public void StartDevMode()
        //{

        //    GSState.NodeParser.StartDevMode();
        //    PTLState.NodeParser.StartDevMode();
        //}
    }
}
